Source/WebCore/html/HTMLLabelElement.cpp

```diff
@@ -73,18 +73,38 @@ Ref<HTMLLabelElement> HTMLLabelElement::create(Document& document)
 
 RefPtr<HTMLElement> HTMLLabelElement::control() const
 {
-    auto& controlId = attributeWithoutSynchronization(forAttr);
-    if (controlId.isNull()) {
+    if (!hasAttributeWithoutSynchronization(forAttr)) {
         // Search the children and descendants of the label element for a form element.
         // per http://dev.w3.org/html5/spec/Overview.html#the-label-element
         // the form element must be "labelable form-associated element".
-        for (const auto& labelableElement : descendantsOfType<HTMLElement>(*this)) {
-            if (labelableElement.isLabelable())
-                return const_cast<HTMLElement*>(&labelableElement);
+        for (const HTMLElement& descendant : descendantsOfType<HTMLElement>(*this)) {
```

Comment:
```suggestion
        for (const auto& descendant : descendantsOfType<HTMLElement>(*this)) {
```

Reply:
Similarly, would you be able to help me understand the reasoning behind preferring less type information here?

Reply:
Ditto. Here, `descendantsOfType<HTMLElement>` tells us exactly what type the iterator will iterate over. There is no point in repeating that information again in the variable declaration.